tsconfig.json

```diff
@@ -0,0 +1,67 @@
+{
+  "compilerOptions": {
+    /* Basic Options */
+    // "incremental": true,                   /* Enable incremental compilation */
+    "target": "es5" /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', or 'ESNEXT'. */,
+    "module": "commonjs" /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */,
+    // "lib": [],                             /* Specify library files to be included in the compilation. */
+    // "allowJs": true,                       /* Allow javascript files to be compiled. */
+    // "checkJs": true,                       /* Report errors in .js files. */
+    // "jsx": "preserve",                     /* Specify JSX code generation: 'preserve', 'react-native', or 'react'. */
+    // "declaration": true,                   /* Generates corresponding '.d.ts' file. */
+    // "declarationMap": true,                /* Generates a sourcemap for each corresponding '.d.ts' file. */
+    // "sourceMap": true,                     /* Generates corresponding '.map' file. */
```

Comment:
Maybe enable source maps (I suggest with the `inlineSourceMap` option) so that assertion failures for tests show Typescript line numbers and are easier to follow.

Reply:
Please don't inline source maps, frontend will be sad.
If no .js files will end up in the frontend (currently we only import `BatchExchange.json` through dex-js ), then `inlineSourceMap` is ok

Reply:
I don't think any of these JS files would get added to the front end with how things are currently imported. `index.js` just imports the contracts ABIs and the encoding methods as of right now.

Also, just curious, what are the issues with inlined source maps with the front end? Don't they get stripped on minification? Also, shouldn't JS bundling tools also work with transitive source maps (as in the minifier would produce a source map, which would map the minified code to the original source by reading the intermediate source map)?

Reply:
No issue with minification, bundling. It works exactly as you described

There are some seemingly random issues when Chrome and/or Firefox devtools sometimes can't read sourcemaps properly and debugging 3rd-party libraries gets difficult.